doc/application/index.rst

```diff
@@ -1178,316 +1185,10 @@ documentation `runningcmake`_ .
 
 .. _runningcmake: http://cmake.org/runningcmake/
 
-.. _application_configuration:
-
-Application Configuration
-*************************
-
-.. _application_kconfig:
-
-Kconfig Configuration
-=====================
-
-The initial configuration for an application is produced by merging
-configuration settings from three sources:
-
-1. A :makevar:`BOARD`-specific configuration file, stored in
-   :file:`boards/ARCHITECTURE/BOARD/BOARD_defconfig` in the Zephyr base
-   directory.
-
-2. Any CMakeCache entries that are prefixed with :makevar:`CONFIG_`.
-
-3. One or more application-specific configuration files.
-
-The application-specific configuration file(s) can be specified in any of the
-following ways. The simplest option is to just have a single :file:`prj.conf`
-file.
-
-1. If :makevar:`CONF_FILE` is set in :file:`CMakeLists.txt` (**before including
```

Comment:
You have removed a bunch of text from these sections, not sure if *too* much. I found some of these tips to be very useful. Also, should we keep a minimal description of Kconfig and its usage in this .rst file? @mbolivar @SebastianBoe

Reply:
What tips are you thinking of? Note that a bunch of stuff has just been rephrased to be shorter or moved around.

Reply:
> Also, should we keep a minimal description of Kconfig and its usage in this .rst file? @mbolivar @SebastianBoe

I did not read the diff in full, but the existing pointers to the new doc that were left in the application guide seem OK to me.

Reply:
Added back a short Kconfig section to the Application page that just mentions `prj.conf` and links off to the Kconfig docs, and linked that one instead from the other Application sections. Should make it harder to miss.

Reply:
@carlescufi ping